### PR TITLE
[quantization] Support PPL Evaluation of Qwen3-VL With Wikitext2 Dataset

### PR DESCRIPTION
**Pull request by dvsav**
# What

This commit adds support for PPL (perplexity) evaluation on the wikitext2 dataset for Qwen3-VL models in `quantize_qwen3_vl_with_gptq.py`.

# Why

PPL metric is listed in issue [PTQ Evaluation — Qwen3-VL](https://github.sec.samsung.net/one-project/TICO/issues/192).

# Implementation Details

## `tico/quantization/evaluation/vlm_eval_utils.py`

1. **Added wikitext2 dataset record to DATASETS dictionary:**
   - `default_split`: "test"
   - `adapter`: None (text-only dataset)
   - `candidates`: ["wikitext"]
   - `config`: "wikitext-2-raw-v1"

2. **Updated `get_dataset()` function:**
   - Added support for optional `config` parameter for datasets requiring a config name
   - Updated dataset loading to use `load_dataset(path=name, name=config, ...)` when config is present
   - Updated log message to include config info

3. **Added `evaluate_ppl()` function:**
   - Accepts: model, tokenizer, dataset, device, stride, max_seq_len, show_progress
   - Concatenates all text from dataset using `"\n\n".join(ds["text"])`
   - Encodes text with tokenizer
   - Calls `perplexity()` from `tico/quantization/wrapq/utils/metrics.py`
   - Returns PPL score as float

## `tico/quantization/wrapq/examples/quantize_qwen3_vl_with_gptq.py`

1. **Added import:** `evaluate_ppl` from vlm_eval_utils

2. **Added CLI arguments:**
   - `--ppl_dataset`: Text dataset for PPL evaluation (choices: ["wikitext2"])
   - `--ppl_stride`: Sliding window stride (default: 512)

3. **Added PPL evaluation for original model** (after MMLU evaluation):
   - Loads dataset via `get_dataset(args.ppl_dataset, n=args.nsamples_for_evaluation)`
   - Calls `evaluate_ppl()` with model, tokenizer, dataset, device, stride, max_seq_len
   - Prints "Original PPL: XX.XX"

4. **Added PPL evaluation for quantized model** (after MMLU evaluation):
   - Same process as original model evaluation
   - Prints "Quantized PPL: XX.XX"

## `tico/quantization/wrapq/wrappers/qwen_vl/quant_for_conditional_generation.py`

1. **Added loss computation in `forward()` method:**
   - When `labels` is provided and `return_dict` is True
   - Calls `self.module.loss_function(logits, labels, vocab_size)`
   - Returns loss in `Qwen3VLCausalLMOutputWithPast`
   - This is required for PPL calculation which uses the model's loss output

# Usage Example

```bash
$ python tico/quantization/wrapq/examples/quantize_qwen3_vl_with_gptq.py \
   --model=Qwen/Qwen3-VL-4B-Instruct \
   --cache_dir=/home/d.savchenkov/models/qwen3-vl-4b \
   --trust-remote-code \
   --gptq_mse=mse \
   --no_GPTQ \
   --ppl_dataset=wikitext2 \
   --nsamples_for_evaluation=50 \
   --embedding_weight_bits=16 \
   --vision_patch_embed_weight_bits=16 \
   --linear_weight_bits=16 \
   --lm_head_weight_bits=16 \
   --nsamples_for_qcalibration=10 \
   --verbose
```

# Example Output

```
=== PPL Evaluation (Original Model) ===
[info] Loaded dataset: wikitext (config=wikitext-2-raw-v1, test, streaming), size=50
Original PPL: 8.38

=== PPL Evaluation (Quantized Model) ===
[info] Loaded dataset: wikitext (config=wikitext-2-raw-v1, test, streaming), size=50
Quantized PPL: 9.40
```

### CODE REVIEW

File: tico/quantization/wrapq/wrappers/qwen_vl/quant_for_conditional_generation.py [hunk 1]
```diff
@@ -143,8 +143,16 @@ def forward(
         )
 
         if return_dict:
+            loss = None
```
Comment (mhs4670go) [inline tico/quantization/wrapq/wrappers/qwen_vl/quant_for_conditional_generation.py]: For consistency with llama, how about this one?

```python
loss = None
if labels is not None:
    loss = self.module.loss_function(
        logits=logits,
        labels=labels,
        vocab_size=self.module.config.text_config.vocab_size,
        **kwargs,
    )

if not return_dict:
    output = (logits,) + outputs[1:]
    return (loss,) + output if loss is not None else output

output = Qwen3VLCausalLMOutputWithPast(
    loss=loss,
    logits=logits,
    past_key_values=outputs.past_key_values,
    hidden_states=outputs.hidden_states,
    attentions=outputs.attentions,
    rope_deltas=outputs.rope_deltas,
)
```

Comment (dvsav) [reply tico/quantization/wrapq/wrappers/qwen_vl/quant_for_conditional_generation.py]: 👍 done

**Review by mhs4670go — APPROVED** [review]
LGTM
